metricbeat/helper/kubernetes/ktest/ktest.go

```diff
@@ -33,9 +33,15 @@ func getFiles(folder string) ([]string, error) {
 	if err != nil {
 		return nil, err
 	}
-	for i, e := range entries {
-		files[i] = filepath.Join(folder, e.Name())
+	fileIndex := 0
+	for _, e := range entries {
+		// this file is only needed for testdata folder
+		if e.Name() != "docs.plain" {
+			files[fileIndex] = filepath.Join(folder, e.Name())
```

Comment:
We do the fileIndex logic, because otherwise we will have an empty entry inside the array. And that empty entry will count as if the filename is ` ` (empty string), and all the tests fail. That is why I had to do it like this @MichaelKatsoulis

Reply:
if you use append method to files array only in case of `if e.Name() != "docs.plain" ` , you will have an empty entry?

Reply:
Yes, the empty entry comes from the array initialization above.

Reply:
The test output goes like this that way:
```
...
ptest.go:96: open _meta/test/..expected: no such file or directory
```

Reply:
The reason for this is that you create a slice with a fixed length of `len(entries)` but then the actual length is `len(entries)-1` because you don't want to include the `docs.plain`. So in this case having a set length of the slice is not a good option.
When I made the comment for having a preallocation of the files slice I didn't know that we may actually get less entries in the slice.

I believe the best option here is to just initialise files as a 0 length slice like `files := make([]string, 0)` and then use append.

Reply:
> files := make([]string, 0)

Initializes an array with length 0. 

If the current approach is bad, I can also iterate over entries first and check if `docs` file exists, and determine the length through that.

Reply:
But going over `entries` twice looks like a bad approach

Reply:
I updated the code to follow this approach